pkg/report/genericsbom.go

```diff
@@ -129,11 +147,41 @@ func (gsbmw GsbomWriter) Write(report types.Report) error {
 	}
 	return nil
 }
+
+func getMetaData(report types.Report) Metadata {
+	metadata := Metadata{}
+	if report.Metadata.RepoTags != nil {
+		metadata["repo_tag"] = report.Metadata.RepoTags[0]
+	}
+	if report.Metadata.RepoDigests != nil {
+		metadata["repo_digest"] = report.Metadata.RepoDigests[0]
+	}
+	return metadata
+}
+
+func getDependencies(results []types.Result, pkg ftypes.Package) []string {
+	var dependencies []string
+	for _, result := range results {
+		for _, dep := range result.Dependencies {
+			if dep.ID == pkg.ID {
```

Comment:
I think this condition can be true only once, so you can `return dep.DependsOn` inside the if statement instead of appending it to a dependencies list